router/batchrouter/asyncdestinationmanager/bing-ads/bulk_uploader.go

```diff
@@ -86,172 +90,124 @@ func (b *BingAdsBulkUploader) Upload(destination *backendconfig.DestinationT, as
-		resp = common.PollStatusResponse{
-			Success:        true,
-			StatusCode:     200,
-			HasFailed:      false,
-			HasWarning:     false,
-			FailedJobsURL:  "",
-			WarningJobsURL: "",
-			OutputFilePath: "",
+	case "FileUploaded", "InProgress", "PendingFileUpload":
+		return common.PollStatusResponse{
+			InProgress: true,
+			StatusCode: 200,
 		}
-		// needs to be retried to fetch the final status
-		statusCode = 500
-		return resp, statusCode
-	} else {
-		/* if uploadStatusResp.RequestStatus == "Failed" || uploadStatusResp.RequestStatus == "UploadFileRowCountExceeded" || uploadStatusResp.RequestStatus == "UploadFileFormatNotSupported" {
-		 */
-		resp = common.PollStatusResponse{
-			Success:        false,
-			StatusCode:     400,
-			HasFailed:      true,
-			HasWarning:     false,
-			FailedJobsURL:  "",
-			WarningJobsURL: "",
-			OutputFilePath: "",
+	default:
+		return common.PollStatusResponse{
+			HasFailed:  true,
+			StatusCode: 400,
 		}
-		statusCode = 400
 	}
-	return resp, statusCode
 }
 
-func (b *BingAdsBulkUploader) Poll(pollInput common.AsyncPoll) (common.PollStatusResponse, int) {
+func (b *BingAdsBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
 	fmt.Println("Polling Bing Ads")
 	var cumulativeResp common.PollStatusResponse
-	var statusCode int
+	var completionStatus []bool
+	var failedJobsInfo string
+	var cumulativeCompletionStatus, cumulativeProgressStatus, cumulativeFailureStatus bool
+	// var statusCode int
 	requestIdsArray := common.GenerateArrayOfStrings(pollInput.ImportId)
 	for _, requestId := range requestIdsArray {
-		resp, status := b.PollSingleImport(requestId)
-		if status != 200 {
+		resp := b.PollSingleImport(requestId)
+		if resp.StatusCode != 200 {
+			// if any of the request fails then the whole request fails
 			cumulativeResp = resp
-			statusCode = status
 			break
 		}
-		cumulativeResp = common.PollStatusResponse{
-			Success:        resp.Success,
-			StatusCode:     200,
-			HasFailed:      cumulativeResp.HasFailed || resp.HasFailed,
-			HasWarning:     cumulativeResp.HasWarning || resp.HasWarning,
-			FailedJobsURL:  "",
-			WarningJobsURL: "",
-			OutputFilePath: cumulativeResp.OutputFilePath + resp.OutputFilePath + ",",
-		}
-		statusCode = status
+		/*
+			Cumulative Response logic:
+			1. If any of the request is in progress then the whole request is in progress and it should retry
+			2. if all the requests are completed then the whole request is completed
+			3. if any of the requests are completed with errors then the whole request is completed with errors
+			4. if any of the requests are failed then the whole request is failed and retried
+			5. if all the requests are failed then the whole request is failed
+		*/
+		completionStatus = append(completionStatus, resp.Complete)
+		cumulativeCompletionStatus = generateCompletionStatus(completionStatus)
+		failedJobsInfo = cumulativeResp.FailedJobsInfo + resp.FailedJobsInfo + ","
```

Comment:
better to append to array and then use strings.Join

Reply:
done